internal/context.go

```diff
@@ -23,21 +23,11 @@ import (
 
 // -- Connector ID ----------------------------------------------------------
 
-type connectorIDCtxKey struct{}
+type ConnectorIDCtxKey struct{}
 
 // ContextWithConnectorID wraps ctx and returns a context that contains the connector ID.
 func ContextWithConnectorID(ctx context.Context, connectorID string) context.Context {
-	return context.WithValue(ctx, connectorIDCtxKey{}, connectorID)
-}
-
-// ConnectorIDFromContext fetches the connector ID from the context. If the
-// context does not contain a connector ID it returns an empty string.
-func ConnectorIDFromContext(ctx context.Context) string {
-	connectorID := ctx.Value(connectorIDCtxKey{})
-	if connectorID != nil {
-		return connectorID.(string) //nolint:forcetypeassert // only this package can set the value, it has to be a string
-	}
-	return ""
+	return context.WithValue(ctx, ConnectorIDCtxKey{}, connectorID)
 }
```

Comment:
Personally, I'd prefer if we leave the struct private and simply call `internal.ConnectorIDFromContext` in the introduced function. Then we have the usage of the context key struct nicely encapsulated and both setter and getter are here in the same place. My 2 cents 🤷

Reply:
i.e. like it was [here](https://github.com/ConduitIO/conduit-connector-sdk/pull/174/commits/5936fbca685ec76e5a34bd794d968a5d0f79c16d)? I don't mind either way.

Reply:
Yes, feels cleaner compared to exposing the context key type IMO.

Reply:
Done :+1: